tracing-attributes/src/expand.rs

```diff
@@ -49,8 +50,34 @@ pub(crate) fn gen_function<'a, B: ToTokens + 'a>(
 
     let warnings = args.warnings();
 
+    let block = if let ReturnType::Type(_, return_type) = &output {
+        let return_type = erase_impl_trait(return_type);
+        // Install a fake return statement as the first thing in the function
+        // body, so that we eagerly infer that the return type is what we
+        // declared in the async fn signature.
+        let fake_return_edge = quote_spanned! {return_type.span()=>
+            #[allow(unreachable_code)]
+            if false {
+                let __tracing_attr_fake_return: #return_type = panic!();
+                return __tracing_attr_fake_return;
+            }
```

Comment:
nit, take it or leave it: maybe worth including the comment in the quoted block, so that it's present in `cargo expand` output and people don't wonder what this code is doing?

Reply:
I don't think that comments show up in `cargo expand` of quoted output, but I did adjust the panic message so that it's being communicated at least in some form.